src/BLSPublicKeyCompendium.sol

```diff
@@ -75,6 +84,24 @@ contract BLSPublicKeyCompendium is IBLSPublicKeyCompendium {
                             VIEW FUNCTIONS
     *******************************************************************************/
 
+    /**
+     * @notice Returns the pubkey of an operator, verifying that the pubkey and its hash are valid
+     * @dev Reverts if the operator has not registered a valid pubkey
+     */
+    function getRegisteredPubkey(address operator) public view returns (BN254.G1Point memory) {
+        BN254.G1Point memory pubkey = operatorToPubkey[operator];
+        bytes32 pubkeyHash = operatorToPubkeyHash[operator];
+
+        require(
+            pubkeyHash != bytes32(0) && BN254.hashG1Point(pubkey) == pubkeyHash, 
+            "BLSPublicKeyCompendium.getRegisteredPubkey: operator is not registered"
+        );
```

Comment:
the second part of this check feels like overkill / not the right place to be enforcing this?
we are already enforcing that `BN254.hashG1Point(operatorToPubkey[operator]) == operatorToPubkeyHash[operator]`
I'm not opposed to some redundant checks in general, but I think a specific benefit of this storage change could be _not_ needing to repeatedly do the `hashG1Point` operation, which this seems to be doing unnecessarily

Reply:
Agree that consistency checks between the mapping should be enforced at write time, not read time. Is there a way they can diverge?

Reply:
see [bb4011e](https://github.com/Layr-Labs/eigenlayer-middleware/pull/51/commits/bb4011ede8aa3e0ee29133d38487530f12806056)